bindings/Sofa/src/SofaPython3/Sofa/Core/Binding_Node.cpp

```diff
@@ -235,6 +235,115 @@ void setFieldsFromPythonValues(Base* self, const py::kwargs& dict)
     }
 }
 
+py::object add( Node* self, const std::string& type, const py::kwargs& kwargs)
+{
+    std::string name {};
+    if (kwargs.contains("name"))
+    {
+        name = py::str(kwargs["name"]);
+        if (sofapython3::isProtectedKeyword(name))
+            throw py::value_error("Cannot call addObject with name " + name + ": Protected keyword");
+    }
```

Comment:
Shouldn't we consider to add a name generation mechanism (with +1) to avoid having name duplicated.